openedx_authz/tests/test_enforcement.py

```diff
@@ -573,3 +580,77 @@ def test_wildcard_library_access(self, scope: str, expected_result: bool):
             "expected_result": expected_result,
         }
         self._test_enforcement(self.POLICY, request)
+
+
+@pytest.mark.django_db
+@ddt
+class StaffSuperuserAccessTests(CasbinEnforcementTestCase):
+    """
+    Tests for staff and superuser automatic permission grants via custom_check.
+
+    This test class verifies that staff members and superusers are automatically
+    granted access to ContentLibrary scopes through the check_custom_conditions function,
+    without requiring explicit role assignments.
+    """
+
+    # Empty policy - no role assignments for staff/superuser users
+    POLICY = []
+
+    def setUp(self) -> None:
+        """Set up the test environment."""
+        super().setUp()
+        User.objects.create_user(username="staff_user", email="staff@example.com", password="test", is_staff=True)
+        User.objects.create_superuser(username="superuser", email="super@example.com", password="test")
+        User.objects.create_user(username="regular_user", email="regular@example.com", password="test")
+
+    @data(
```

Comment:
Do we need a test that handles non-existent library keys? I assume that would return False, right?

Reply:
Yes! I added a test case here: https://github.com/openedx/openedx-authz/pull/114/commits/780569819f962cefbd0435a1ea5ce531aea94eda